cirq-core/cirq/testing/consistent_act_on.py

```diff
@@ -46,12 +46,15 @@ def state_vector_has_stabilizer(state_vector: np.ndarray, stabilizer: DensePauli
     """
 
     qubits = LineQubit.range(protocols.num_qubits(stabilizer))
+    complex_dtype: Type[np.complexfloating] = np.complex64
+    if np.issubdtype(state_vector.dtype, np.complexfloating):
+        complex_dtype = cast(Type[np.complexfloating], state_vector.dtype)
     args = state_vector_simulation_state.StateVectorSimulationState(
         available_buffer=np.empty_like(state_vector),
         qubits=qubits,
         prng=np.random.RandomState(),
         initial_state=state_vector.copy(),
-        dtype=state_vector.dtype,
+        dtype=complex_dtype,
```

Comment:
in this case, since this is in the simulation direct call path, should we just do a cast here?

Reply:
Are we sure `state_vector.dtype` is complex?
If so we can make it simpler with cast.